regions/core/bounding_box.py

```diff
@@ -10,43 +10,49 @@
 
 class BoundingBox(object):
     """
-    Rectangular bounding box in integer pixel indices (not floats).
+    A rectangular bounding box in integer (not float) pixel indices.
 
     Parameters
     ----------
     ixmin, ixmax, iymin, iymax : int
-        The bounding indices.
-        Note that the upper values (iymax and ixmax) are
-        exclusive as for normal slices in Python.
+        The bounding box pixel indices.  Note that the upper values
+        (``iymax`` and ``ixmax``) are exclusive as for normal slices in
+        Python.  The lower values (``ixmin`` and ``iymin``) must not be
+        greater than the respective upper values (``ixmax`` and
+        ``iymax``).
 
     Examples
     --------
-
     >>> from regions import BoundingBox
-    # Constructing a BoundingBox like this is cryptic:
+    # constructing a BoundingBox like this is cryptic:
     >>> bbox = BoundingBox(1, 10, 2, 20)
-    # Better use keyword arguments for readability:
+    # it's better to use keyword arguments for readability:
     >>> bbox = BoundingBox(ixmin=1, ixmax=10, iymin=2, iymax=20)
     >>> bbox  # nice repr, useful for interactive work
     BoundingBox(ixmin=1, ixmax=10, iymin=2, iymax=20)
-    # sometimes it's useful to check if two bboxes are the same
+    # sometimes it's useful to check if two bounding boxes are the same
     >>> bbox == BoundingBox(ixmin=1, ixmax=10, iymin=2, iymax=20)
     True
     >>> bbox == BoundingBox(ixmin=99, ixmax=10, iymin=2, iymax=20)
     False
-    # Shape and slices can be useful when working with cutout numpy arrays
+    # "shape" and "slices" can be useful when working with numpy arrays
     >>> bbox.shape  # numpy order: (y, x)
     (18, 9)
     >>> bbox.slices  # numpy order: (y, x)
     (slice(2, 20, None), slice(1, 10, None))
-    # Extent is useful when plotting bbox with matplotlib
+    # "extent" is useful when plotting the BoundingBox with matplotlib
     >>> bbox.extent  # matplotlib order: (x, y)
     (0.5, 9.5, 1.5, 19.5)
     >>> print(bbox.as_patch())
     Rectangle(0.5,1.5;9x18)
     """
 
     def __init__(self, ixmin, ixmax, iymin, iymax):
+        if ixmin > ixmax:
```

Comment:
Should ``ixmin`` be strictly less than ``ixmax`` since the upper limit is exclusive? (that is, the smallest possible box is if ``ixmax == ixmin + 1``.